src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -45,6 +46,19 @@ export type SurvivalType = {
 export class StudyViewPageStore {
 
     constructor() {
+        reaction(
```

Comment:
not sure I understand what this does.

Reply:
I need to hide charts initially when the page is loaded depending on number of categories(slices) each clinical attribute has.

Reply:
this needs a lot of comments.  if i understand correctly, you want to determine what charts to show initially based on the initial response of the clinicalData service, but NOT on subsequent?  if that's the case there may be a much simpler way to accomplish this.  in general we should almost never use reactions/autorun unless really necessary

Reply:
Yes, that's true. I'll add comments and see if I can get rid of reaction block

Reply:
a different strategy might be, make an observable property on store, say, initialData=undefined.    Then give it a setter than only ever updatest that value if it is undefined.    this.intitialData = this.initialData || newData;

Reply:
would that work?

Reply:
I'll give it a try